docs/DeveloperGuide.md

```diff
@@ -99,28 +112,99 @@ Optionally, you can follow the [Using Checkstyle](UsingCheckstyle.md) document t
  These configuration information are used by:
     - `GitDownloader` to determine which repository to download from and which branch to check out to.
     - `AuthorshipReporter` and `CommitsReporter` to determine the range of commits and files to analyze.
     - `ReportGenerator` to determine the directory to output the report.
 
 
 ## HTML Dashboard
+The source files for the dashboard is located in [`frontend/src`](../frontend/src) and is built by [spuild](https://github.com/ongspxm/spuild2) before being packaged into the JAR file to be extracted as part of the report.
+
+The main HTML file is generated from [`frontend/src/index.jade`](../frontend/src/index.jade).
+
+[Vue](https://vuejs.org/v2/api/) (pronounced /vjuː/, like view) is a progressive framework for building user interfaces. It is heavily ultilized in the dashboard to dynamically update the information in the various views. (Style guide available [here](https://vuejs.org/v2/style-guide/)).
+
+![dashboard screenshot](images/dashboard.png)
+
+### Dashboard Architecture
+![dashboard architecture](images/dashboard-architecture.png)
+
+The main Vue object (`window.app`) is responsible for the loading of the dashboard (through `summary.json`). Its `repos` attribute is tied to the global `window.REPOS`, and is passed into the various other modules when the information is needed.
+
+`window.app` is broken down into two main parts
+- the summary view
+- and the tabbed interface
+
+Summary view act as the main dashboard which shows the various calculations. </br>
+Tabbed interface is responsible for loading various modules such as authorship to display additional information.
+
+### Javascript Files
+- [**main.js**](../frontend/src/static/js/main.js) - main controller that pushes content into different modules
+- [**api.js**](../frontend/src/static/js/api.js)- loading and parsing of the dashboard content
+- [**v_summary.js**](../frontend/src/static/js/v_summary.js) - module that supports the ramp chart view
+- [**v_authorship.js**](../frontend/src/static/js/v_authorship.js) - module that supports the authorship view
+
+### JSON report files
+- **summary.json** - a list of all the repositories and their respective details
+- **projName/commits.json** - contains information of the users' commits information (e.g. line deletion, insertion, etc), grouped by date
+- **projName/authorship.json** - contains information from git blame, detailing the author of each line for all the processed files
+
+### Main (main.js)
+This contains the logic for main VueJS object, `window.app`, which is responsible for passing the necessary data into the relevant modules to be loaded.
+
+`v_summary` and `v_authorship` are components which will be embedded into dashboard and will render the corresponding content based on the data passed into it from the main `window.app`.
```

Comment:
actually i mean a diagram under each module. the architecture is too vague.

Reply:
more drawings? okay :cry:

Reply:
if you don't have powerpoint etc... can do paper drawing, we try to convert it to diagram for you.